internal/config.go

```diff
@@ -62,6 +64,8 @@ func InitConfig(configName string) error {
 		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
 			return err
 		}
+	} else {
```

Comment:
We could print outside the `else` to reduce the a bit the complexity, as it if doesn't return here, the result is positive.
Like here: https://github.com/spf13/viper#reading-config-files

Reply:
I can make it like 
```
err := viper.ReadInConfig()

if err != nil {
	if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
		return err
	}
}

log.Infof("Using config file: %s", viper.ConfigFileUsed())
```

Then it happens that we have that line logged also in a scenario where we do not use any config file (none explicitly provided, none found in the expected locations)

That would print
```
time="2021-11-15 17:27:56" level=info msg="Using config file: "
```

Otherwise I'd need another if to check whether we have something in ` viper.ConfigFileUsed()`

wdyt?

Reply:
Or I can make a function like this

```
func usedConfigFile() string {
	configFile := viper.ConfigFileUsed()
	if configFile == "" {
		return "none"
	}
	return configFile
}
```

and use it like 
```
log.Infof("Using config file: %s", usedConfigFile())
```

That would print
```
msg="Using config file: none"

or

msg="Using config file: /etc/trento/agent.yaml"

or whatever the provided file is
```

Preferences?

Reply:
I would just add
```go
if viper.ConfigFileUsed() != nil {
  log.Infof(......
}
```

Reply:
Thanks @fabriziosestito for bringing me back to earth :joy: 
Fixed that way